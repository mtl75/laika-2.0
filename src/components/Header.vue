<template>
  <div class="header-template">
    <b-nav class="starter-nav" tabs fill>
      <b-nav-item class="laika-title" data-test="laika_title" disabled>
        <b>Laika: A Space Adventure</b>
      </b-nav-item>
      <b-nav-item disabled>
        <b-progress data_test='progress_bar' :value="correctPrompts" :max="10" show-progress animated></b-progress>
      </b-nav-item>
    </b-nav>
  </div>
</template>
<script>
export default {
  props: {
    correctPrompts: Number,
    totalPrompts: Number
  },
  data: function() {
    return {};
  }
};
</script>
