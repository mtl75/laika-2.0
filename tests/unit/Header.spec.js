import { createLocalVue } from '@vue/test-utils'
import Header from '@/components/Header.vue';
import { shallowMount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'

const localVue = createLocalVue()

// install plugins as normal
localVue.use(BootstrapVue)



describe ('Header', () => {
    const wrapper= shallowMount(Header)
    test('renders correctly', () => {
        const wrapper = shallowMount(Header)
        expect(wrapper.element).toMatchSnapshot()
      })
  
    it('Displays title Text', () =>{
        expect(wrapper.find('[data-test="laika_title"]').text()).toEqual(
            'Laika: A Space Adventure')
        })
 })
