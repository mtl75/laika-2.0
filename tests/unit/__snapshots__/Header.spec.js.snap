// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  class="header-template"
>
  <b-nav
    class="starter-nav"
    fill=""
    tabs=""
  >
    <b-nav-item
      class="laika-title"
      data-test="laika_title"
      disabled=""
    >
      <b>
        Laika: A Space Adventure
      </b>
    </b-nav-item>
     
    <b-nav-item
      disabled=""
    >
      <b-progress
        animated=""
        data_test="progress_bar"
        max="10"
        show-progress=""
      />
    </b-nav-item>
  </b-nav>
</div>
`;
